### DISCUSSION TITLE
[Edge Functions Runtime] waitUntil blocking and limits

### DISCUSSION THREAD

**Discussion by ShantanuNair** [vercel/community#2789] [Help]
### Summary

Two questions when using waitUntil to process a task, with the intention being it is run after the response and the function exits whenever this completes:
1.  Does waitUntil block the response? I mean, if I have a webhook and want to respond to it with a 200 and then want to process it after sending the response, can I do that?
2. Does waitUntil have a cloud flare worker like 30s timeout on edge functions? In the above example, assume my webhook process has to wait a long wall clock time for third party APIs to respond (say OpenAI), but uses less that the CPU Time limit, then can it run for 30 minutes comfortably, without being killed like in cloud flare workers (which has a 30s timeout post-response)?

A question trying to figure out the same was raised in https://github.com/orgs/vercel/discussions/2513, but the response provided was vague and not addressing the particular details.

### Steps to Reproduce

N/A

**Comment by smaeda-ks** [comment]
answered in: https://github.com/orgs/vercel/discussions/2513#discussioncomment-6128504

> Does waitUntil block the response? I mean, if I have a webhook and want to respond to it with a 200 and then want to process it after sending the response, can I do that?

No, it does not block the response.

**Reply by ShantanuNair** [reply]
@smaeda-ks I see, thanks! Regarding the previous discussion you just answered in about the roughly 30s limit, please consider adding that to the docs, like cloud flare has. I hadn't gotten into the weeds of edge functions and so was trying to figure out if CPU Time alone was a limit in Verccel's Edge runtime.
